docker/build/mule.ubuntu.Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM docker:19 as docker
+FROM python:3.7
+
+COPY --from=docker /usr/local/bin/docker /usr/local/bin/docker
+COPY *.yaml ./
+
+RUN pip install mulecli
+
+RUN apt-get update && apt-get install -y build-essential curl libboost-all-dev autoconf bsdmainutils
+RUN curl https://dl.google.com/go/go1.12.linux-amd64.tar.gz | tar -xzf -
+RUN mv go /usr/local
+ENV GOROOT=/usr/local/go \
+    GOPATH=$HOME/go
+ENV PATH=$GOPATH/bin:$GOROOT/bin:$PATH
```

Comment:
This isn't great.  We should make the ticket to install go from only one location a priority.

Reply:
That's **probably** all that is needed, but I'd rather keep it in sync with the version that we're using internally.

Reply:
I believe this actually installs 1.12.0. We probably want to use 1.12.17 instead. Why are we making an ubuntu image with go installed on it though? I thought we decided to do our builds on the centos images

Reply:
This is just a placeholder so I could have a version of go and keep moving forward.  My point in calling it out is that it shouldn't be here at all, we should be referencing a common place that is doing the installation.

This isn't doing a build, it's testing, and I needed python3.7 and mule to do that.  I actually took most of this from your open PR.

I frankly don't care too much what dockerfile I use, as long as it gets the job done.

Reply:
Just to follow up here, I mis-labeled the dockerfile, it's actually a Debian distro:
```
root@f88dec6be829:~# cat /etc/*release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
I like this dockerfile for tasks where I need `mule`, but again, I'm open to better proposals and am not particularly concerned as to the dockerfile.